src/event/ServerSocket.cxx

```diff
@@ -35,6 +35,7 @@
 #include "util/Domain.hxx"
 #include "util/ScopeExit.hxx"
 #include "Log.hxx"
+#include "config/Path.hxx"
```

Comment:
I neither like pulling in big deps but I had no better idea.
We need someting like ParsePath() there for home dirs (only allowing absolute paths would not be nice)...
And I am not sure if its a good idea to move AddGeneric away from ServerSocket.
What do you suggest?

Reply:
I had a similar idea like your commit a while ago, but didn't have a quick idea how to solve this dependency problem. Resolving home directories must not be done here because that may require accessing configuration data from a global variable; during startup, MPD may still run as "root", but "~/" must resolve to the configured user's home directory, not to root's.
Maybe the config library needs a function which expands a string which is possibly a path, but leaves it as UTF-8 string (instead of using the FS charset / `class Path`). But then it may need to convert the configured UTF-8 string to FS charset, expand the path, convert back to UTF-8, and then the ServerSocket converts it to FS charset again. Ugly!
Another idea: implement a function in the config library which uses a configured (UTF-8) string to set up a given `ServerSocket` instance. No change to `ServerSocket` needed, and the heavy-dependency code lives in the config library.